libs/vm-modules/benchmark/model_charge/model_with_charge.cpp

```diff
@@ -0,0 +1,768 @@
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "math/tensor.hpp"
+#include "vectorise/fixed_point/fixed_point.hpp"
+
+#include "vm_modules/math/tensor.hpp"
+#include "vm_modules/ml/model/model.hpp"
+#include "vm_modules/ml/model/model_estimator.hpp"
+#include "vm_modules/vm_factory.hpp"
+
+#include "gmock/gmock.h"
+
+#include <sstream>
+
+#include "benchmark/benchmark.h"
+
+#include <vector>
+
+namespace vm_modules {
+namespace benchmark {
+namespace ml {
+namespace modell {
+
+namespace {
+
+using VMPtr = std::shared_ptr<fetch::vm::VM>; 
+using SizeType     = fetch::math::SizeType;
+using SizeRef = fetch::math::SizeType const &;
+using StringPtrRef = fetch::vm::Ptr<fetch::vm::String> const &;
+
+//////////////////////////
+// VM Objects factories //
+//////////////////////////
+
+VMPtr NewVM()
+{
+  using VMFactory = fetch::vm_modules::VMFactory;
+
+  // setup the VM
+  auto module = VMFactory::GetModule(fetch::vm_modules::VMFactory::USE_SMART_CONTRACTS);
+  auto vm          = std::make_shared<fetch::vm::VM>(module.get());
+
+  return vm;
+}
+
+fetch::vm::Ptr<fetch::vm::String> vmString(VMPtr &vm, std::string const &str)
+{
+  return fetch::vm::Ptr<fetch::vm::String>{new fetch::vm::String{vm.get(), str}};
+}
+
+fetch::vm::Ptr<fetch::vm_modules::math::VMTensor> vmTensor(VMPtr &vm,
+                                                    std::vector<SizeType> const &shape)
+{
+  return vm->CreateNewObject<fetch::vm_modules::math::VMTensor>(shape);
+}
+
+fetch::vm::Ptr<fetch::vm_modules::ml::model::VMModel> vmSequentialModel(VMPtr &vm)
+{
+  auto model_category = vmString(vm, "sequential");
+  auto model = vm->CreateNewObject<fetch::vm_modules::ml::model::VMModel>(model_category);
+  return model;
+}
+
+fetch::vm::Ptr<fetch::vm_modules::ml::model::VMModel> vmSequentialModel(VMPtr &vm, std::vector<SizeType> &sizes, std::vector<bool> &activations)
+{
+  if(sizes.size() != (activations.size()+1))
+  {
+    throw std::runtime_error{"Wrong configuration for multilayer VMModel"};
+  }
+
+  auto model = vmSequentialModel(vm);
+  auto size = activations.size();
+  auto layer_type = vmString(vm, "dense");
+  auto activation_type = vmString(vm, "relu");
+
+  for (std::size_t i{0}; i < size; ++i)
+  {
+    auto input_size =  sizes[i];
+    auto output_size = sizes[i+1];
+
+    if (activations[i])  // TOFIX if NOT
```

Comment:
cryptic fix message?

Reply:
these TOFIXs are points I need to discuss with Jiri